gprofiler/profilers/java.py

```diff
@@ -938,8 +941,6 @@ def _profile_process(self, process: Process, duration: int, spawned: bool) -> Pr
             self._profiled_pids.add(process.pid)
 
         logger.info(f"Profiling{' spawned' if spawned else ''} process {process.pid} with async-profiler")
-        app_metadata = self._metadata.get_metadata(process)
```

Comment:
This was the correct place for app metadata & appid retrieval, I don't want them to be retrieved before the safety checks are done (async-profiler loaded, JVM profiling supported etc). Bring them back and move the container name here as well?